src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -667,6 +709,17 @@ internal override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArgu
             }
         }
 
+        private static bool? DecodeMaybeNullWhenOrNotNullWhenAttribute(AttributeDescription description, CSharpAttributeData attribute, DiagnosticBag diagnostics)
+        {
+            var arguments = attribute.CommonConstructorArguments;
+            if (arguments.Length == 1)
+            {
+                var arg = arguments[0];
+                return arg.DecodeValue<bool>(SpecialType.System_Boolean);
```

Comment:
>arg.DecodeValue<bool>(SpecialType.System_Boolean) [](start = 23, length = 49)

It doesn't look like we would get a ```null``` value if ```DecodeValue``` fails due to one reason or another.